packages/shell-api/src/custom-inspect.ts

```diff
@@ -0,0 +1,55 @@
+import type { InspectOptions, inspect as _inspect } from 'util';
+
+const customInspectSymbol = Symbol.for('nodejs.util.inspect.custom');
+
+function customDocumentInspect(
+  this: Document,
+  depth: number,
+  inspectOptions: InspectOptions,
+  inspect: typeof _inspect
+) {
+  const newInspectOptions = {
+    ...inspectOptions,
+    depth: Infinity,
+    maxArrayLength: Infinity,
+    maxStringLength: Infinity,
+  };
+
+  // reuse the standard inpect logic for an object without causing infinite
+  // recursion
+  const copyToInspect: any = Array.isArray(this) ? this.slice() : { ...this };
+  delete copyToInspect[customInspectSymbol];
+  return inspect(copyToInspect, newInspectOptions);
+}
+
+function addInspectSymbol(obj: any) {
+  if (!(obj as any)[customInspectSymbol]) {
+    Object.defineProperty(obj, customInspectSymbol, {
+      value: customDocumentInspect,
+      enumerable: false,
+      writable: true,
+      configurable: true,
+    });
+  }
+}
+
+export function addCustomInspect(obj: any) {
+  if (Array.isArray(obj)) {
+    addInspectSymbol(obj);
+    for (const item of obj) {
+      addCustomInspect(item);
+    }
+  } else if (
```

Comment:
Maybe better to reuse an isSimpleObject() test like https://github.com/mongodb-js/devtools-shared/blob/ff08cc5e55212890f1d8635f545474201b1e1167/packages/mongodb-redact/src/utils.ts